app/tests/src/cardano-mnemonic-crypto-provider.js

```diff
@@ -156,6 +156,83 @@ describe('wallet addresses derivation', () => {
   })
 })
 
+describe('checking input integrity', () => {
+  it('should properly check unspent output coin amount with raw input transaction data', () => {
+    const rawTxBody = Buffer.from(
```

Comment:
I would declare the rawTxBody outside of the 'it' blocks if it is reused. I would also declare 4 separate inputs - input1Right, input2Right, input1WrongTxHash, input1WrongCoins and combine them to arrays in the test cases below which would improve the readability of the tests in case they would need to be rewritten.